H2TauTau/python/proto/analyzers/LHEWeightAnalyzer.py

```diff
@@ -0,0 +1,181 @@
+from PhysicsTools.Heppy.analyzers.core.Analyzer import Analyzer
+from PhysicsTools.Heppy.analyzers.core.AutoHandle import AutoHandle
+import PhysicsTools.HeppyCore.framework.config as cfg
+from DataFormats.FWLite import Handle, Runs
+from ROOT.gen import WeightsInfo
+from CMGTools.RootTools.statistics.TreeNumpy import TreeNumpy
+from ROOT import TFile, TH1F
+
+class LHEWeightAnalyzer( Analyzer ):
+    """Read the WeightsInfo objects of the LHE branch and store them
+       in event.LHE_weights list.
+
+       If the WeightsInfo.id is a string, replace it with an integer.
+
+       So far the only allowed string format is "mg_reweight_X",
+       which gets stored as str(10000+int(X))
+
+       If w.id is an unknown string or anything but a string or int,
+       a RuntimeError is raised.
+    """
+    def __init__(self, cfg_ana, cfg_comp, looperName ):
+        super(LHEWeightAnalyzer,self).__init__(cfg_ana,cfg_comp,looperName)
+        self.index = []
+
+    def declareHandles(self):
+        super(LHEWeightAnalyzer, self).declareHandles()
+        self.mchandles['LHEweights'] = AutoHandle('externalLHEProducer',
+                                                  'LHEEventProduct',
+                                                  mayFail=True,
+                                                  fallbackLabel='source',
+                                                  lazy=False )
+
+    def beginLoop(self, setup):
+        super(LHEWeightAnalyzer,self).beginLoop(setup)
+
+        self.run = Runs(self.cfg_comp.files[0])
+        runsHandle = Handle('LHERunInfoProduct')
+        self.run.getByLabel('externalLHEProducer', runsHandle)
+
+
+        self.scale = TH1F('scale', 'scale', 10, 0, 10)
+        self.pdf = TH1F('pdf', 'pdf', 101, 0, 101)
+        self.alpha_s = TH1F('alpha_s', 'alpha_s', 2, 0, 2)
+
+
+        iterator = runsHandle.product().headers_begin()
+
+        # scan iterator so that it includes weight information
+        for it in range(100):
+            flag = False
+            for itt in iterator.base().lines():
+                print itt
+                if itt.find('weight id') != -1: flag = True
+
+            if flag == True: break
```

Comment:
avoid comparisons with True
best:
if flag:
    break

if you really want to check identity use "if flag is False"